static/js/compose_closed_ui.js

```diff
@@ -1,35 +1,113 @@
+import * as narrow_state from "./narrow_state";
+import * as people from "./people";
 import * as popovers from "./popovers";
+import * as recent_topics from "./recent_topics";
 
-export function hide_reply_button() {
-    $(".reply_button_container").hide();
-    $("#compose_buttons").css("justify-content", "flex-end");
+function update_stream_button(btn_text, title) {
+    $("#left_bar_compose_stream_button_big").text(btn_text);
+    $("#left_bar_compose_stream_button_big").prop("title", title);
 }
 
-export function show_reply_button() {
-    $(".reply_button_container").show();
-    $("#compose_buttons").css("justify-content", "flex-start");
+function update_conversation_button(btn_text, title) {
+    $("#left_bar_compose_private_button_big").text(btn_text);
+    $("#left_bar_compose_private_button_big").prop("title", title);
+}
+
+function update_buttons(text_stream) {
+    const title_stream = text_stream + " (c)";
+    const text_conversation = $t({defaultMessage: "New private message"});
+    const title_conversation = text_conversation + " (x)";
+    update_stream_button(text_stream, title_stream);
+    update_conversation_button(text_conversation, title_conversation);
+}
+
+export function update_buttons_for_private() {
+    const text_stream = $t({defaultMessage: "New stream message"});
+    update_buttons(text_stream);
+}
+
+export function update_buttons_for_stream() {
+    const text_stream = $t({defaultMessage: "New topic"});
+    update_buttons(text_stream);
+}
+
+export function update_buttons_for_recent_topics() {
+    const text_stream = $t({defaultMessage: "New stream message"});
+    update_buttons(text_stream);
+}
+
+function set_reply_button_label(label) {
+    $(".compose_reply_button_label").text(label);
+}
+
+export function set_standard_text_for_reply_button() {
+    set_reply_button_label($t({defaultMessage: "Compose message"}));
 }
 
 export function update_reply_recipient_label(message) {
-    message = message || message_lists.current.selected_message();
+    if (message === undefined) {
+        if (message_lists.current.empty()) {
+            // For empty narrows where there's a clear reply target,
+            // i.e. stream+topic or a single PM conversation, we label
+            // the button as replying to the thread.
+            if (narrow_state.narrowed_to_topic()) {
+                message = {
+                    stream: narrow_state.stream(),
+                    topic: narrow_state.topic(),
+                };
+            } else if (narrow_state.narrowed_to_pms_conversation_or_less()) {
+                // TODO: This is a total hack.  Ideally, we'd rework
+                // this to not duplicate the actual compose_actions.js
+                // logic for what happens when you click the button,
+                // and not call into random modules with hacky fake
+                // "message" objects.
+                const user_ids = people.user_ids_string_to_ids_array(narrow_state.pm_string());
```

Comment:
We are facing a puppeteer failure here, because if `narrow_state.pm_string()` is `undefined`  `people.user_ids_string_to_ids_array` will cause `split` error because it only expects string.


Might be the fix for this will be (tested ✅ ),
```diff
- else if (narrow_state.narrowed_to_pms_conversation_or_less()) {
+ else if (narrow_state.narrowed_to_pms_conversation_or_less() && narrow_state.pm_string()) {
```

Reply:
Which narrow is it looking at when it throws the exception?  The bug could be in `narrow_state.pm_string`.

(I would have expected that any narrow with a `pm-with` operator would have `narrow_state.pm_string()` not be undefined).  

We should figure out the view and print-debug.

Reply:
Oh, it's an invalid narrow, isn't it?

Reply:
OK, I think we can actually skip `narrowed_to_pms_conversation_or_less` and just call `narrow_state.pm_string` here, given that finding.  Thanks for debugging this!